lib/reline/key_stroke.rb

```diff
@@ -99,6 +108,37 @@ def expand(input)
 
   private
 
+  # returns match status of CSI/SS3 sequence and matched length
+  def match_unknown_escape_sequence(input)
+    idx = 0
+    esc = 27 # "\e".ord
+    return [:unmatched, nil] unless input[idx] == esc
+    idx += 1
+    idx += 1 if input[idx] == esc
+
+    case input[idx]
+    when nil
+      return [:matching, nil]
+    when 91 # == '['.ord
+      # CSI sequence
+      idx += 1
+      idx += 1 while idx < input.size && (0x30..0x3f).cover?(input[idx])
+      idx += 1 while idx < input.size && (0x20..0x2f).cover?(input[idx])
+      input[idx] ? [:matched, idx + 1] : [:matching, nil]
+    when 79 # == 'O'.ord
+      # SS3 sequence
+      input[idx + 1] ? [:matched, idx + 2] : [:matching, nil]
```

Comment:
F1, F2, F3, F4 key in macOS `["\eOP", "\eOQ", "\eOR", "\eOS"]`

There are many other type of escape sequence, but I think it's CSI and SS3 is enough for keyboard input sequence. some other sequence ends with `"\a"` or "`\e\\`" (example: `"\e]0;NewWindowTitle\e\\"`) but if we implement it and user mistakenly input ESC+], irb looks hanging up.